### DISCUSSION TITLE
Extend colors from external file

### DISCUSSION THREAD

**Discussion by ines-b** [tailwindlabs/tailwindcss#11579] [Help]
Hi everyone,

I was just wondering whether or not it was possible to extend colors in `tailwind.config.js` file with an external json file?
It would be quite convenient for me to share a file with custom colors between several projects, and to extend tailwind's default color with the ones contained in that file.
I didn't find anything in that regard in the documentation!

Thanks a lot, 
Cheers

**Comment by mybearworld** [comment]
It's just a JavaScript file, so you can just read the JSON file with JavaScript and put it in.

**Reply by ines-b** [reply]
I've already tried that, but for some reason, my nuxt app cannot start, it fails with a `Javascript heap out of memory` error.
I'm using node v16.19.1.

Is there any size restriction on the amount of custom colors/values that be can be used to extend tailwind's default configuration?

**Reply by ines-b** [reply]
I eventually switched to a JS file that a I export as a module, and I managed to make it work for 1 color (at the moment).
But with my whole design system (~400 colors), my app crashes on boot.

**Reply by amirnsamimi** [reply]
have you found any solution on this ?
